src/iff.imageio/iffoutput.cpp

```diff
@@ -144,28 +145,57 @@ IffOutput::open(const std::string& name, const ImageSpec& spec, OpenMode mode)
     //    FOR4 <size> TBMP
     // Tiles:
     //       RGBA <size> tile pixels
+    //       ZBUF <size> tile pixels
     //       RGBA <size> tile pixels
-    //       RGBA <size> tile pixels
+    //       ZBUF <size> tile pixels
     //       ...
 
     // saving 'name' and 'spec' for later use
     m_filename = name;
 
-    if (!check_open(mode, spec, { 0, 8192, 0, 8192, 0, 1, 0, 4 }))
+    if (!check_open(mode, spec, { 0, 8192, 0, 8192, 0, 1, 0, 5 }))
         return false;
     // Maya docs say 8k is the limit
 
+    // validate supported formats: RGB (3), RGBA (4), RGBAZ (5)
+    if (spec.nchannels < 3 || spec.nchannels > 5) {
+        errorfmt(
+            "Cannot write IFF file with {} channels (only RGB, RGBA, RGBAZ supported)",
+            spec.nchannels);
+        return false;
+    }
+
+
+    // IFF image files only supports UINT8 and UINT16.  If something
+    // else was requested, revert to the one most likely to be readable
+    // by any IFF reader: UINT8
+
+    TypeDesc base_format = spec.format;
+    if (base_format != TypeDesc::UINT8 && base_format != TypeDesc::UINT16) {
+        errorfmt("Unsupported format {}. Converting to UINT8.", base_format);
+        base_format = TypeDesc::UINT8;
+    }
```

Comment:
I think that most of the readers pick the supported data type that loses the least precision versus the one requested. So if they ask for something at least as big as uint16, we should probably give them uint16 rather than uint8, which is even less fidelity.